### ISSUE TITLE
Suggestion: flag suspicious impicit string concatenation

### ISSUE THREAD

**Issue by abgros** [astral-sh/ruff#23939]
### Summary

Consider this code:
```py
LIST_OF_STRINGS = [
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
]

print(len(LIST_OF_STRINGS))
```

When you run this code, the output is `4`, because of the missing comma. This mistake is easy to miss due to the strings being relatively long. I'm opening this issue after being caught out for the second time by a missing comma.

I suggest using the heuristic that impicit string concatenation is "suspicious" if it's used within a list or tuple where there are elements besides the concatenated string.

**Comment by ntBre** [comment]
This example should be caught by the new [implicit-string-concatenation-in-collection-literal (ISC004)](https://docs.astral.sh/ruff/rules/implicit-string-concatenation-in-collection-literal/#implicit-string-concatenation-in-collection-literal-isc004) rule, which is still in preview:

https://play.ruff.rs/815f8e8f-3b72-4359-a9b4-9aaf40891b14

**Comment by abgros** [comment]
@ntBre Thanks, someone on Discord had let me know that this exact idea seems to already exist... well, for now I've just enabled preview lints and found many of them to be very good. There ought to be a lint for projects that don't have preview lints enabled.
